helm/resource_release.go

```diff
@@ -953,10 +953,45 @@ func resourceDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{})
 
 	// Set desired version from the Chart metadata if available
 	if len(chart.Metadata.Version) > 0 {
-		return d.SetNew("version", chart.Metadata.Version)
+		err := d.SetNew("version", chart.Metadata.Version)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := d.SetNewComputed("version")
+		if err != nil {
+			return err
+		}
 	}
 
-	return d.SetNewComputed("version")
+	desiredValues, err := getValues(d)
+	if err != nil {
+		return err
+	}
+	cloakSetValues(desiredValues, d)
+	desiredValuesJson, err := json.Marshal(desiredValues)
+	if err != nil {
+		return err
+	}
+	desiredValuesJsonString := string(desiredValuesJson)
+
+	if desiredValuesJsonString != d.Get("metadata.0.values").(string) &&
+		!(desiredValuesJsonString == "{}" && d.Get("metadata.0.values").(string) == "null") {
+		err = d.SetNew("metadata", []map[string]interface{}{{
```

Comment:
Terraform [does not support to `SetNew` with nested key](https://github.com/hashicorp/terraform-plugin-sdk/issues/459) so I set whole metadata here.